### PR TITLE
Allow featureGroup initialization with no overlay corners

### PR DESCRIPTION
**Pull request by sashadev-sky**
A step to fix https://github.com/publiclab/mapknitter/issues/919 (<=== Add issue number here)
- #335 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by sashadev-sky** [comment]
@rexagod @jywarren this PR is very small in code changes but makes a bigger difference to how the library can be used. 

`corners` has never been a required option for `L.distortableImageOverlay`, but they were for `DistortableCollection` because I  added a call to `img.editing.enable`  from there. I moved this call inside the overlay class instead, since we already have an event listener there for  image `load`.

I updated the README pretty thoroughly to basically make the point that for single image interface you don't really need them but for multi you want to include them or the images will render on top of eachother. The `index.html` and `select.html` demos now reflect this too! 

I think it actually makes the plugin a lot easier to use, because people can add images to the map quickly, and use our API to retrieve the corners for a position they like and then add in the corners option. Otherwise they have to calculate it themselves and risk distorting their image.

**Comment by sashadev-sky** [comment]
in MK, hopefully this will allow us to stop adding images to the map instead of the feature group.

It also calls `img.editing.enable` by default once they're added to the map, so I bumped to `0.8.0`. but this is already what we were doing for the feature group anyway.

**Comment by sashadev-sky** [comment]
ok I just found that @rexagod uses custom logic for `img.editing.enable` #312 and I have a feeling others might want to as well, so I am going  to create an `enabled` option that can be set to false so that image editing isn't automatically enabled on load

**Comment by jywarren** [comment]
OK, interesting, so will this then need some more work before reviewing it?
Thanks!

**Comment by jywarren** [comment]
Ah, does this fix https://github.com/publiclab/mapknitter/issues/919 or what else would be required? I'm very eager to get that resolved, so any tips appreciated here! Thank you!

**Comment by sashadev-sky** [comment]
@jywarren Might fix it or might just be one step in fixing it. It's basically ready for review, all I am doing is adding an `enabled` option which will run based on a conditional. I will finish this ASAP. The main part to really review is to just read through the README.

**Comment by jywarren** [comment]
Ok, looks good to me! I rushed in a fix for the legacy exporter just now to
address the urgent issue and I hope that'll fix it. But still eager to see
this merged!

**Comment by jywarren** [comment]
As soon as it's complete, let's bump version in Mapknitter to get this
pulled in.

**Comment by sashadev-sky** [comment]
I had to do a lot of restructuring and make a #335 for this option to work properly with both groups. almost done tho!

**Comment by sashadev-sky** [comment]
@jywarren ok this turned into a complicated one but it is done now. @rexagod @ViditChitkara I think `enable`, `disable`, `editable` API could come up for you guys so check it out! Let me know your thoughts.

Easiest probably to just look at README and try the code out if u'd like

**Comment by ViditChitkara** [comment]
That's awesome!! Will check this out.
Thanks

File: README.md [hunk 1]
```diff
@@ -192,33 +183,23 @@ When instantiating a Distortable Image, pass in a `fullResolutionSrc` option set
...
 
 This option will override other options related to the toolbar, such as [`selected: true`](#Selected)
 
-## Setup - Multiple Image Interface
-
-1.  From the root directory, run `npm install` or `sudo npm install`
+## Multiple Image Interface
 
-2.  Open `examples/select.html` in a browser (todo -- add gh pages demo)
+(todo -- add gh pages demo)
```
Comment (jywarren) [inline README.md]: Oh, does select.html not work for this anymore?

File: README.md [hunk 2]
```diff
@@ -303,66 +294,60 @@ Currently it supports multiple image selection and translations, and WIP we are
...
 Defaults:
 
--   **Exports**
+* **Exports**
 
--   **Deletes (<kbd>delete</kbd>, <kbd>backscpace</kbd>)**
+* **Deletes (<kbd>delete</kbd>, <kbd>backspace</kbd>)**
 
-    - Permanently deletes groups of selected images from the map.
+  * Permanently deletes groups of selected images from the map.
```
Comment (jywarren) [inline README.md]: Should we note it has a `confirm()` just in case?

**Comment by jywarren** [comment]
And shall we bump to `0.7.1` with this? Any breaking changes? I think not...

File: src/DistortableImageOverlay.js [hunk 3]
```diff
@@ -146,12 +161,15 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
...
+
     if (edit.toolbar && edit.toolbar instanceof L.DistortableImage.PopupBar) {
       edit._updateToolbarPos();
     }
+
+    return this;
   },
 
-  _setCornersFromPoints: function(pointsObj) {
+  setCornersFromPoints: function(pointsObj) {
```
Comment (rexagod) [inline src/DistortableImageOverlay.js]: 🎉

File: README.md (hunk 1 above)
Comment (sashadev-sky) [reply README.md]: oh haha it does that was silly of me I'll remove the todo

File: README.md (hunk 2 above)
Comment (sashadev-sky) [reply README.md]: yes will do

File: src/DistortableImageOverlay.js (hunk 3 above)
Comment (sashadev-sky) [reply src/DistortableImageOverlay.js]: 🎉I will add that one to the API soon.

File: src/edit/DistortableCollection.Edit.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    try {
+      if (!this.toolbar) {
+        this.toolbar = L.distortableImage.controlBar({
+          actions: this.editActions,
+          position: 'topleft'
+        }).addTo(map, this);
+        this.fire('toolbar:created');
+      }
+    } catch (e) { }
```
Comment (rexagod) [inline src/edit/DistortableCollection.Edit.js]: I guess JS was throwing an error for `this.toolbar` not being initialized for this if that's why we have this in a try block? If that's the case then why not check "in" the object (`this`) keys for the `toolbar` property? That should avoid the try catch blocks and keep things simple.

Also, is suppressing the errors here okay? Maybe we could send these to the error logs? Just a thought, what do you think?

**Comment by sashadev-sky** [comment]
> And shall we bump to `0.7.1` with this? Any breaking changes? I think not...

I made it `0.8.0` because a) seemed like a big API update and I wanted people to notice the update corners stuff.  b) Could be breaking for any implementations that make the image editing interface turn on by some custom logic rather than load. But doesn't matter anyway and the odds are slim. 

Updated to `0.7.1`

File: src/edit/DistortableCollection.Edit.js (hunk 4 above)
Comment (sashadev-sky) [reply src/edit/DistortableCollection.Edit.js]: @rexagod valid point but thats not really new code so if we can  open a new issue and discuss it there that would be great! 

Honestly not even sure what the issue was anymore it was a while ago. you can try rewriting and see if it works?

Comment (rexagod) [reply src/edit/DistortableCollection.Edit.js]: Sure! I'll look into this.

**Pull request merged**